pkg/render/apiserver.go

```diff
@@ -651,7 +651,7 @@ func (c *apiServerComponent) calicoCustomResourcesClusterRole() *rbacv1.ClusterR
 			},
 		},
 	}
-	if c.cfg.KubernetesVersion == nil || !(c.cfg.KubernetesVersion != nil && c.cfg.KubernetesVersion.Major < 2 && c.cfg.KubernetesVersion.Minor < 30) {
+	if c.cfg.KubernetesVersion == nil || !(c.cfg.KubernetesVersion != nil && c.cfg.KubernetesVersion.Major < 2 && c.cfg.KubernetesVersion.Minor >= 30) {
```

Comment:
Can we make sure we have unit tests that verify the correct behavior for relevant versions?

e.g., 1.29, 1.30, 1.31

Reply:
I am not familiar with the codebase yet and was hoping i'd learn how to run the tests via CI check logs. 

I'll spend some time understanding the tests.